### PR TITLE
En 5565 unit tests for header signature verify

### PR DESCRIPTION
**Pull request by miiu96**

### CODE REVIEW

File: process/headerCheck/headerSignatureVerify_test.go [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+	args.NodesCoordinator = nodesCoordinator
+	hdrSigVerifier, _ := NewHeaderSigVerifier(args)
+	header := &dataBlock.Header{}
+
+	err := hdrSigVerifier.VerifyRandSeed(header)
+	require.Nil(t, err)
+	require.True(t, wasCalled)
+}
+
+func TestHeaderSigVerifier_VerifyRandSeedShouldErr(t *testing.T) {
```
Comment (bogdan-rosianu) [inline process/headerCheck/headerSignatureVerify_test.go]: rename to `TestHeaderSigVerifier_VerifyRandSeedShouldErrWhenVerificationFails`

File: process/headerCheck/headerSignatureVerify_test.go [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+
+	args := createHeaderSigVerifierArgs()
+	hdrSigVerifier, _ := NewHeaderSigVerifier(args)
+	header := &dataBlock.Header{}
+
+	err := hdrSigVerifier.VerifyRandSeedAndLeaderSignature(header)
+	require.Equal(t, sharding.ErrNilRandomness, err)
+}
+
+func TestHeaderSigVerifier_VerifyRandSeedAndLeaderSignatureVerifyShouldErr(t *testing.T) {
```
Comment (bogdan-rosianu) [inline process/headerCheck/headerSignatureVerify_test.go]: rename to `TestHeaderSigVerifier_VerifyRandSeedAndLeaderSignatureVerifyShouldErrWhenValidationFails`

File: process/headerCheck/headerSignatureVerify_test.go [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+	hdrSigVerifier, _ := NewHeaderSigVerifier(args)
+	header := &dataBlock.Header{
+		PubKeysBitmap: []byte("11"),
+	}
+
+	err := hdrSigVerifier.VerifySignature(header)
+	require.Equal(t, ErrWrongSizeBitmap, err)
+}
+
+func TestHeaderSigVerifier_VerifySignatureNotEnoughtSigsShouldErr(t *testing.T) {
```
Comment (bogdan-rosianu) [inline process/headerCheck/headerSignatureVerify_test.go]: Enough*

File: process/headerCheck/headerSignatureVerify_test.go (hunk 1 above)
Comment (miiu96) [reply process/headerCheck/headerSignatureVerify_test.go]: done.

File: process/headerCheck/headerSignatureVerify_test.go (hunk 2 above)
Comment (miiu96) [reply process/headerCheck/headerSignatureVerify_test.go]: done.

File: process/headerCheck/headerSignatureVerify_test.go (hunk 3 above)
Comment (miiu96) [reply process/headerCheck/headerSignatureVerify_test.go]: done.

**Comment by bogdan-rosianu** [comment]
Still no test for the constructor

**Pull request merged**